intercom-provider/src/main/java/com/busybusy/intercom_provider/IntercomProvider.java

```diff
@@ -156,13 +172,33 @@ public void endTimedEvent(@NonNull AnalyticsEvent timedEvent) throws IllegalStat
 
 	private void logIntercomEvent(@NonNull AnalyticsEvent event) throws IllegalStateException
 	{
-		// guard clause
+		Map<String, Object> sanitizedAttributes;
 		if (event.getAttributes() != null && event.getAttributes().keySet().size() > MAX_METADATA_ATTRIBUTES)
 		{
-			throw new IllegalStateException("Intercom does not support more than " + MAX_METADATA_ATTRIBUTES +
-					                                " metadata fields. See https://docs.intercom.com/the-intercom-platform/track-events-in-intercom.");
+			if (this.quietMode)
+			{
+				sanitizedAttributes = new LinkedHashMap<>();
+				int count = 0;
+				for (String key : event.getAttributes().keySet())
```

Comment:
I missed the Kotlin collections functions here so badly! :persevere: 

I am contemplating how I can convert this library's code to Kotlin, but I'm wondering what kind of DSL if any we should use.

Reply:
Java is so bad compared to Kotlin lol. This could be done in Kotlin for half the amount of code. 
If you have time to convert the project to Kotlin I'm all for it. It's up to you.
As for using a DSL. You could have a simple DSL for consuming this library like
```
AnalyticsKit {
      registerProvider {
          provider1
      }
      registerProvider {
          provider2
      }
  }
```

Reply:
Thanks, this is a great suggestion!. I'll keep this in mind.